src/transport/SecureSessionMgr.cpp

```diff
@@ -164,15 +168,22 @@ CHIP_ERROR SecureSessionMgr::SendMessage(SecureSessionHandle session, PayloadHea
     VerifyOrExit(admin != nullptr, err = CHIP_ERROR_INCORRECT_STATE);
     localNodeId = admin->GetNodeId();
 
-    if (payloadHeader.HasMessageType(Protocols::SecureChannel::MsgType::MsgCounterSyncReq) ||
-        payloadHeader.HasMessageType(Protocols::SecureChannel::MsgType::MsgCounterSyncRsp))
+    if (IsControlMessage(payloadHeader))
     {
         packetHeader.SetSecureSessionControlMsg(true);
     }
 
+    if (!IsControlMessage(payloadHeader))
+    {
+        // The peer counter should be already initialized during CASE session establishment
+        // It may not true for group messages. But this assert will quickly catch one's attention.
+        assert(state->GetSessionMessageCounter().GetPeerMessageCounter().IsSynchronized());
```

Comment:
I don't really understand this assert as regards groups.  This whole operation makes no sense for group messages; what does "peer" even mean there?

Is this really trying to assert "this is not a group message, because we don't know how to check for this there"?  Then why not just assert that this is not a group message?  Past that, why would we care whether the peer counter is synced when we _send_?

Reply:
The code is here regarding to the spec:

```
Message synchronization to a peer is initiated when a reliable message is to be transmitted to a destination node whose message counter is unknown
````

Which is a case only for group message.

Since we are going to remove this statement, I'll remove these codes.